engine/collection/compliance/core.go

```diff
@@ -105,6 +109,9 @@ func NewCore(
 // OnBlockProposal handles incoming block proposals.
 // No errors are expected during normal operation.
 func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.ClusterBlockProposal) error {
+	startTime := time.Now()
+	defer c.hotstuffMetrics.BlockProcessingDuration(time.Since(startTime))
```

Comment:
⚠️ Arguments for deferred functions are evaluated right away, so this will always report ~0s.

Reply:
https://github.com/onflow/flow-go/pull/3623/commits/7078c9cc5b14a6299357f809d50b3162c9091e3e